services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/api/containers_extension.py

```diff
@@ -204,9 +210,14 @@ async def push_output_ports(
     port_keys = [] if port_keys is None else port_keys
 
     await send_message(rabbitmq, f"Pushing outputs for {port_keys}")
-    await nodeports.upload_outputs(
-        mounted_volumes.disk_outputs_path, port_keys=port_keys
-    )
+    try:
+        await nodeports.upload_outputs(
+            mounted_volumes.disk_outputs_path, port_keys=port_keys
+        )
+    except NodeNotFound as err:
+        detail = dict(error="node_not_found", message=f"{err}")
+        raise HTTPException(status.HTTP_404_NOT_FOUND, detail=detail) from err
```

Comment:
shouldn't you also add this to the list of responses above? thisone is 404 but above is 422!??